### DISCUSSION TITLE
Are subsequent pages also crawl-able (SEO) when we change route in NextJs application?

### DISCUSSION THREAD

**Discussion by mustafam-globant** [vercel/next.js#24345] [Help]
I understand that when a NextJs page is visited, server(SSR) renders and send HTML along with necessary javascript to hydrate page on client side. This make page SEO friendly.

But when we change route (say "/about"), it does not actually send HTML file but just about.js (Content-Type: application/javascript) file (and looking into file it looks like React VirtualDOM, but am not sure). And this get painted on DOM and we see about page.

<img width="1790" alt="Screenshot 2021-04-22 at 1 01 39 PM" src="https://user-images.githubusercontent.com/64200846/115688265-a126a700-a378-11eb-8483-23b03460844f.png">

So then, my question is -

- Is this about page SEO-friendly? (because it's not HTML file)
- If not, then does it mean only initial page is good for SEO and not subsequent routes (in case of Client side routing)?

Thank you

**Comment by akellbl4** [comment]
Hi,
This behavior is specific for client-side navigation.
If you open `/about` in a new tab, it will be sent from the server as HTML.

**Reply by mustafam-globant** [reply]
Yes, I get that opening in new tab HTML will be sent. My question is for this specific client-side navigation. In this case there is no SEO for about page right?

**Reply by akellbl4** [reply]
As @techrome wrote, I have the same opinion on crawlers.

**Reply by mustafam-globant** [reply]
Ok, thanks. I understand that it doesn't matter how client-side navigation works. What matters is how all the pages of application are sent back as response when visited directly isn't?

**Reply by akellbl4** [reply]
Yes and I can add more.
If the crawler is able to render the next page with JS it's able to crawl it. It won't affect SEO.
Otherwise, it will act as a browser with disabled JS, and every subsequent page will be opened as HTML.
You can try to disable JS in your browser and navigation will work just fine and in this case, you will get HTML for every next page.

**Comment by techrome** [comment]
Search crawlers, as far as I know, open every new URL in a new tab so your every page will be crawled the same.

**Reply by mustafam-globant** [reply]
Ok, so you mean what matters to crawlers is that whether HTML is sent when direct subsequent URL is visited (like /about) directly, right?

**Reply by techrome** [reply]
Yeah, crawlers basically scan every page for `anchor` tags with `href` attribute and then "click" on every link to open in new tab. So as long as your every page is discoverable from other pages and returns HTML when accessing it directly, the crawlers will be able to read all your pages. At least that's how I understand how they work

**Reply by mustafam-globant** [reply]
Thanks!
